test/ctia/http/generative/fulltext_search_test.clj

```diff
@@ -360,3 +362,40 @@
              (-> res first
                  (select-keys (keys expected))
                  (= expected))))))))))
+
+(deftest enforcing-fields-with-feature-flag-test
+  (testing "feature flag set? fields should be enforced"
+    (are [properties query expected-fields]
+         (helpers/with-properties properties
+           (helpers/fixture-ctia-with-app
+            (fn [app]
+              (helpers/set-capabilities!
+               app "foouser" ["foogroup"] "user" (capabilities/all-capabilities))
+              (whoami-helpers/set-whoami-response
+               app "45c1f5e3f05d0" "foouser" "foogroup" "user")
+              (let [results (atom nil)]
+                (with-redefs [crud/make-search-query
+                              (fn [_ q _]
+                                (reset! results
+                                        (->> q :full-text first :fields
+                                             (map keyword)
+                                             set))
+                                (throw (ex-info "Early termination" {})))]
```

Comment:
This should be done through extending the protocol with a mocked fn, but in my attempts to do it the "right way" were not very successful, and since I'm planning to change the way how search-fields are enforced anyway, this entire test block will have to change sometime soon.